lib/Cron/SocialUpdate.php

```diff
@@ -25,18 +25,39 @@
 
 use OCA\Contacts\Service\SocialApiService;
 
+use OCP\AppFramework\Http;
+use OCP\BackgroundJob\IJobList;
+
 class SocialUpdate extends \OC\BackgroundJob\QueuedJob {
 	/** @var SocialUpdateService */
 	private $social;
+	/** @var IJobList */
+	private $jobList;
 
-	public function __construct(SocialApiService $social) {
+	public function __construct(SocialApiService $social,
+								IJobList $jobList) {
 		$this->social = $social;
+		$this->jobList = $jobList;
 	}
 
 	protected function run($arguments) {
 		$userId = $arguments['userId'];
+		$offsetBook = $arguments['offsetBook'];
+		$offsetContact = $arguments['offsetContact'];
```

Comment:
This could be undefined from an old job? In that case add `?? null;`

Reply:
Sorry, I am not sure I understand. I could verify if the book/contact are actually existing and if they don't, reset them. Is that what you mean?

Reply:
Yeah that is what I'm suggesting.
`$a['b'] ?? null` is short for `isset($a['b']) ? $a['b'] : null`

Reply:
I have added this plus some more sanity checks, as the contact or address book that are remembered may be deleted between the execution of two chunks.